subsys/net/lib/coap/coap.c

```diff
@@ -50,9 +50,16 @@ LOG_MODULE_REGISTER(net_coap, CONFIG_COAP_LOG_LEVEL);
 
 #define BASIC_HEADER_SIZE	4
 
+/* CoAP retry backoff factor. A value of 200 means a factor of 2.0 */
+#define COAP_BACKOFF_PERCENT 200
+
 /* The CoAP message ID that is incremented each time coap_next_id() is called. */
 static uint16_t message_id;
 
+static uint8_t max_retransmission;
+static uint32_t ack_timeout;
+static uint16_t coap_backoff_percent;
```

Comment:
If this would be a static struct, you could also initialize it.

```suggestion
static struct coap_config_t coap_configuration = {
   .max_retransmission = CONFIG_COAP_MAX_RETRANSMIT;
   .ack_timeout = CONFIG_COAP_INIT_ACK_TIMEOUT_MS;
   .coap_backoff_percent = COAP_BACKOFF_PERCENT;
  }
```

But same apply here. Initial values don't have to be run-time thing.